### PR TITLE
feat: change hourly rate in invoice item to decimal and required

### PR DESCRIPTION
**Pull request by inpaceee**
closes: #976

### CODE REVIEW

File: packages/backend/src/graphql/scalars/decimal.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { Prisma } from '@progwise/timebook-prisma'
+
+import { builder } from '../builder'
+
+export const DecimalScalar = builder.scalarType('Decimal', {
```
Comment (YohannesTesfay) [inline packages/backend/src/graphql/scalars/decimal.ts:5]: You exported this but didn't use it. My comment on `invoiceItem.ts` on the `hourlyRate` is related to this import.

File: packages/backend/src/prisma/migrations/20241203163728_invoice_item_hourly_rate_decimal/migration.sql [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/*
+  Warnings:
+
+  - You are about to alter the column `hourlyRate` on the `InvoiceItem` table. The data in that column could be lost. The data in that column will be cast from `DoublePrecision` to `Decimal(65,30)`.
+  - Made the column `hourlyRate` on table `InvoiceItem` required. This step will fail if there are existing NULL values in that column.
+
+*/
+-- AlterTable
+ALTER TABLE "InvoiceItem" ALTER COLUMN "hourlyRate" SET NOT NULL,
+ALTER COLUMN "hourlyRate" SET DATA TYPE DECIMAL(65,30);
```
Comment (YohannesTesfay) [inline packages/backend/src/prisma/migrations/20241203163728_invoice_item_hourly_rate_decimal/migration.sql:10]: Isn't 65,30 too big for currency? Have you looked into it?

**Review by YohannesTesfay — CHANGES_REQUESTED** [review]

File: apps/web/frontend/generated/gql/graphql.ts [hunk 3]
```diff
@@ -17,6 +17,7 @@ export type Scalars = {
   Date: string
   /** A date-time string at UTC, such as 2007-12-03T10:15:30Z, compliant with the `date-time` format outlined in section 5.6 of the RFC 3339 profile of the ISO 8601 standard for representation of dates and times using the Gregorian calendar. */
   DateTime: string
+  Decimal: any
```
[the remark this answers is not in the document]
Comment (inpaceee) [reply apps/web/frontend/generated/gql/graphql.ts:20]: Looking at the typing of Int and Float in graphql file, type `number` should be used here, I think

File: packages/backend/src/graphql/invoiceItem/invoiceItem.ts [hunk 4]
```diff
@@ -7,7 +7,11 @@ export const InvoiceItem = builder.prismaObject('InvoiceItem', {
     start: t.expose('start', { type: 'DateTime', nullable: true }),
     end: t.expose('end', { type: 'DateTime', nullable: true }),
     duration: t.exposeInt('duration', { description: 'Duration of the invoice item in minutes' }),
-    hourlyRate: t.exposeFloat('hourlyRate', { description: 'Hourly rate for the invoice item', nullable: true }),
+    hourlyRate: t.expose('hourlyRate', {
+      type: 'Decimal',
```
[the remark this answers is not in the document]
Comment (inpaceee) [reply packages/backend/src/graphql/invoiceItem/invoiceItem.ts:11]: Okay, but you'd have to explain me later why

File: packages/backend/src/graphql/invoiceItem/invoiceItem.ts [hunk 5]
```diff
@@ -7,7 +7,11 @@ export const InvoiceItem = builder.prismaObject('InvoiceItem', {
...
+      description: 'Hourly rate for the invoice item',
+      nullable: true,
```
[the remark this answers is not in the document]
Comment (inpaceee) [reply packages/backend/src/graphql/invoiceItem/invoiceItem.ts:13]: Right, I overlooked it, thanks!

Regarding default nullable behaviour, maybe we did it for better readability?

File: packages/backend/src/prisma/migrations/20241203163728_invoice_item_hourly_rate_decimal/migration.sql (hunk 2 above)
Comment (inpaceee) [reply packages/backend/src/prisma/migrations/20241203163728_invoice_item_hourly_rate_decimal/migration.sql:10]: You're right, it appears to be the default value, will change

**Comment by inpaceee** [comment]
image

**Comment by inpaceee** [comment]
https://github.com/hayes/pothos/commit/5abbd9fc5df4d34a1a019296fb0a636d63d6e4bb#diff-723459fffaf6a960ad20210e8805b74ff095b022ace6be7af12c33eda6742a1bR72

**Review by YohannesTesfay — APPROVED** [review]

**Pull request merged**